File: БР1.1/Данилова Анастасия/labs/rental-platform/shared/rabbitmq/connection.go
package rabbitmq

import (
	"fmt"

	amqp "github.com/rabbitmq/amqp091-go"
)

func Connect(url string) (*amqp.Connection, error) {
	if url == "" {
		return nil, fmt.Errorf("rabbitmq url is empty")
	}
	return amqp.Dial(url)
}

func Channel(conn *amqp.Connection) (*amqp.Channel, error) {
	return conn.Channel()
}

File: БР1.2/Москалец Данила/labs/lab2/sangel-microservices/user-service/event-bus/src/connection.ts
import amqp from 'amqplib';

let connection: amqp.ChannelModel | null = null;
let channel: amqp.Channel | null = null;

export async function getConnection(): Promise<amqp.ChannelModel> {
  if (!connection) {
    const url = process.env.RABBITMQ_URL || 'amqp://localhost:5672';
    connection = await amqp.connect(url);
    console.log(' RabbitMQ connected');
  }
  return connection;
}

export async function getChannel(): Promise<amqp.Channel> {
  if (!channel) {
    const conn = await getConnection();
    channel = await conn.createChannel();
    console.log(' RabbitMQ channel created');
  }
  return channel;
}

export async function closeConnection(): Promise<void> {
  if (channel) {
    await channel.close();
    channel = null;
  }
  if (connection) {
    await connection.close();
    connection = null;
  }
  console.log('🔌 RabbitMQ connection closed');
}